mirrord-layer/src/pod_api.rs

```diff
@@ -49,52 +52,37 @@ struct RuntimeData {
 }
 
 impl RuntimeData {
-    async fn from_k8s(
-        client: Client,
-        pod_name: &str,
-        pod_namespace: &str,
-        container_name: &Option<String>,
-    ) -> Result<Self> {
-        let pods_api: Api<Pod> = Api::namespaced(client, pod_namespace);
-        let pod = pods_api.get(pod_name).await?;
-        let node_name = &pod.spec.unwrap().node_name;
-        let container_statuses = &pod.status.unwrap().container_statuses.unwrap();
-        let container_info = if let Some(container_name) = container_name {
-            &container_statuses
-                .iter()
-                .find(|&status| &status.name == container_name)
-                .ok_or_else(|| {
-                    LayerError::ContainerNotFound(
-                        container_name.clone(),
-                        pod_namespace.to_string(),
-                        pod_name.to_string(),
-                    )
-                })?
-                .container_id
-        } else {
-            info!("No container name specified, defaulting to first container found");
-            &container_statuses.first().unwrap().container_id
-        };
-
-        let container_info = container_info
+    async fn from_k8s(client: Client, target: &str, target_namespace: &str) -> Result<Self> {
+        let target = target.parse::<Target>()?;
+
+        let pods_api: Api<Pod> = Api::namespaced(client.clone(), target_namespace);
+        let deployment_api: Api<Deployment> = Api::namespaced(client.clone(), target_namespace);
+
+        // TODO: figure a way to pass the Api in a generic way, don't want to pass two different
```

Comment:
Why aren't you passing `client` rather than the APIs, then letting `PodData`/`DeploymentData` initialize the API inside `container_info`?

Reply:
thought the same before, that generic thing was a bad design here probably. fixed, thanks!